src/WebJobs.Script/ParseUtility.cs

```diff
@@ -0,0 +1,170 @@
+            }
+            return triggerBindingMetadata;
+        }
+
+        public static void CreateDefaultBindings(FunctionMetadata functionMetadata, BindingMetadata triggerBindingMetadata)
+        {
+            var triggerType = triggerBindingMetadata.Type;
+            if (triggerType.Equals("httptrigger", StringComparison.OrdinalIgnoreCase))
+            {
+                //add in http trigger if they didn't specify it
+                var httpTrigger = functionMetadata.Bindings.FirstOrDefault(
+                    p => p.Type.Equals("httptrigger", StringComparison.OrdinalIgnoreCase) && p.Direction == BindingDirection.In);
+                if (httpTrigger == null)
+                {
+                    httpTrigger = triggerBindingMetadata;
+                    httpTrigger.Direction = BindingDirection.In;
+                    httpTrigger.Name = DefaultHttpTriggerName;
+                    httpTrigger.Type = "httpTrigger";
+                    functionMetadata.Bindings.Add(httpTrigger);
+                }
+                //add in http response if they didn't specify it
+                var response =
+                    functionMetadata.Bindings.FirstOrDefault(
+                        p => p.Type.Equals("http", StringComparison.OrdinalIgnoreCase) && p.Direction == BindingDirection.Out);
+                if (response == null)
+                {
+                    response = new HttpBindingMetadata()
+                    {
+                        Direction = BindingDirection.Out,
+                        Name = DefaultHttpResponseName,
+                        Type = "http"
+                    };
+                    functionMetadata.Bindings.Add(response);
+                }
+            }
+            else if (triggerType.Equals("timertrigger", StringComparison.OrdinalIgnoreCase))
+            {
+                var timerTrigger = functionMetadata.Bindings.FirstOrDefault(
+                    p => (p.Type.Equals("timertrigger", StringComparison.OrdinalIgnoreCase) && p.Direction == BindingDirection.In));
+                if (timerTrigger == null)
+                {
+                    timerTrigger = triggerBindingMetadata;
+                    timerTrigger.Direction = BindingDirection.In;
+                    timerTrigger.Name = DefaultTimerTriggerName;
+                    timerTrigger.Type = "timerTrigger";
+                    functionMetadata.Bindings.Add(timerTrigger);
+                }
+            }
+        }
+    }
+
+    //Schema for the Yaml object
+    public class ApiConfig
+    {      
+        public string Language { get; set; }
+        public TableDetails TableStorage { get; set; }
+        public string CommonCode { get; set; }
+        //supressed to allow yaml parser to assign a value
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
```

Comment:
Typically, the suppressions are added to globally instead on within the code.  You can undo this and when the warning/error shows up again, select the option that adds the suppression in the global file.